### DISCUSSION TITLE
Migrating to a new template location

### DISCUSSION THREAD

**Discussion by AgentK9** [copier-org/copier#1689] [Q&A] [answered]
Hey!

We moved our templates to a new location - everything about the template is the same except for the git repo location. The templates in the old location still exist - is it safe to assume that the best way to tell the existing projects that there's a new location for the templates (in an automated fashion) is via migrations?

Has this use case been covered by copier?

**Comment by pawamoy** [comment]
I don't think the "template does not exist anymore" case is covered. Maybe if the old location redirected to the new one, Copier could detect it (I'm not sure if it's Git or GitHub that notifies it) and update the location in the answers file 🤔

**Comment by yajo** [comment, marked as answer]
When you rename a repo, github and gitlab will create a redirection for it. In general terms, Git will follow that redirection, and Copier will use Git, so it will follow it too.

If I were you, I'd add a migration, yes.

**Reply by AgentK9** [reply]
> When you rename a repo, github and gitlab will create a redirection for it. In general terms, Git will follow that redirection, and Copier will use Git, so it will follow it too.

While this is true, in my case it does not apply. For a few different reasons I'm not moving the project - I'm essentially cloning and pushing to a new remote.

I think that moving a project is best for most users, and migration is second best.
